backend/post/views.py

```diff
@@ -167,21 +200,45 @@ def get_author_id(request):
         yy = xx[1].split("/posts")
         author_id= xx[0]+yy[0]
         return author_id
+    if "followers" in request.build_absolute_uri():
+        xx=request.build_absolute_uri().split('service/')
+        yy = xx[1].split("/followers")
+        author_id= xx[0]+yy[0]
+        return author_id
+    if "friends" in request.build_absolute_uri():
+        xx=request.build_absolute_uri().split('service/')
+        yy = xx[1].split("/friends")
+        author_id= xx[0]+yy[0]
+        return author_id
     else:
         xx=request.build_absolute_uri()[:-7].split('service/')
         author_id= xx[0]+xx[1]
         return author_id
 
-
-    
-def get_post_id(request):
+def get_foreign_id(request):
     xx=request.build_absolute_uri().split('service/')
-    author_id= xx[0]+xx[1]
+    yy = xx[1].split("/followers")
+    author_id= xx[0]+'authors'+yy[1]
     return author_id
 
+def get_friend_id(request):
+    xx=request.build_absolute_uri().split('service/')
+    yy = xx[1].split("/friends")
+    author_id= xx[0]+'authors'+yy[1]
+    return author_id    
+def get_post_id(request):
+    if "likes" in request.build_absolute_uri():
+        xx=request.build_absolute_uri().split('service/')
+        yy = xx[1].split("/likes")
+        author_id= xx[0]+yy[0]
+        return author_id
+    else:
+        xx=request.build_absolute_uri().split('service/')
+        author_id= xx[0]+xx[1]
+        return author_id
+
 def check_author_id(request):
     author_id= get_author_id(request)
     author = Author.objects.filter(id = author_id)
-    print(author.exists())
     return author.exists()
```

Comment:
A couple of things to note here, but I'll leave it as is. I'll make the changes to post once you make your other changes.